jujugui/static/gui/src/app/components/entity-content/entity-content.js

```diff
@@ -32,37 +32,115 @@ YUI.add('entity-content', function() {
     },
 
     /**
+      Generate the list of configuration options for a charm.
+
+      @method _generateCharmConfig
+      @param {Object} entityModel The entity model.
+      @return {Object} The options markup.
+    */
+    _generateCharmConfig: function(entityModel) {
+      var options = entityModel.get('options');
+      if (!options) {
+        return;
+      }
+      var optionsList = [];
+      Object.keys(options).forEach(function(name) {
+        var option = options[name];
+        option.name = name;
+        optionsList.push(
+          <juju.components.EntityContentConfigOption
+            key={name}
+            option={option} />
+        );
+      }, this);
+      return (
+        <dl>
+          {optionsList}
+        </dl>);
+    },
+
+    /**
+      Generate the list of configuration options for a bundle.
+
+      @method _generateBundleConfig
+      @param {Object} entityModel The entity model.
+      @return {Object} The options markup.
+    */
+    _generateBundleConfig: function(entityModel) {
+      var services = entityModel.get('services');
+      if (!services) {
+        return;
+      }
+      // Generate the options for each service in this bundle.
+      var servicesList = Object.keys(services).map((service) => {
+        var options = services[service].options || {};
+        // Generate the list of options for this service.
+        var optionsList = Object.keys(options).map((name, i) => {
+          return (
+            <div className="entity-content__config-option"
+              key={name + i}>
+              <dt className="entity-content__config-name">
+                {name}
+              </dt>
+              <dd className="entity-content__config-description">
+                <p>
+                  {options[name]}
+                </p>
+              </dd>
+            </div>);
+        });
+        var classes = {
+          'entity-content__bundle-config': true
+        };
+        if (optionsList.length === 0) {
+          optionsList.push(
+            <div key="none">
+              No config options for this service.
+            </div>);
+        }
+        return (
+          <juju.components.ExpandingRow
+            classes={classes}
```

Comment:
Unfortunately this is how we have to do it with the way the classNames util expects the object.